main.bal

```diff
@@ -34,7 +34,14 @@ isolated function mapToInternalUrl(string url) returns string {
     // apply all replacements specified in the url map, keys are interpreted as regex
     var replacedUrl = url;
     foreach var key in internalUrlMap.keys() {
-        replacedUrl = regex:replaceFirst(url, key, internalUrlMap.get(key));
+        string strippedKey = key;
+
+        // remove enclosing quotes if necessary
```

Comment:
This should be marked as a workaround for a possible bug with a TODO or FIXME comment